### ISSUE TITLE
Current IPTV add-ons are dead

### ISSUE THREAD

**Issue by BoredLama** [rleroi/stremio-tv#1]
The current IPTV add-ons are now dead:
https://www.reddit.com/r/StremioAddons/comments/aphlza/dead_free_iptv_addon_for_stremio/

There's a lot of info in that post, you can also find the source code for the previous add-ons and the reason their dead.

Also, I see you mentioned `123tv.live` (now called `123tvgo.com`) in this repo's description, those won't work as a remote add-on because their streams would need to be proxied (which for live streams would use up a ton of bandwidth), they do header `referer` (or maybe `user-agent`) checks on the server side.

You can see that because VLC doesn't pass headers without you setting them manually, and the streams from that site don't work in VLC.

**Comment by rleroi** [comment]
Right, I could've known 123tv wouldn't work. Hadn't tried the 123tv site yet.
Initially the addon would use ArconaiTV.us which didn't check the user-agent before, but they started checking it after a few months when I started working on the addon.
Do you know some other IPTV sites which we could try? I was wondering if it is possible to use AceStream streams in Stremio, since it is a torrent based streaming protocol. I'm going to try if this works in Stremio.

**Comment by BoredLama** [comment]
AceStream is a bitch.. although it uses a p2p live streaming protocol, it's one of their own making, the source code is private, and will only work if people have the AceStream Server installed locally.

The AceStream Server does have a local api, although the api is crazy, depending on the operating system and different versions that work on the operating system, the api can be completely different from one system to another.

And even then, it would still need users to install the thing locally, and then maybe a local add-on that converts that `acestream://` protocol to something playable, any other torrent engine won't work with acestream.

So no, I don't see acestream as a viable option unfortunately.

As for websites for free IPTV, `123tvgo.com`, `firstonetv.live` and `arconaitv.us` are also the ones I know to be best.. everything else I looked at didn't seem to be as good..

But even so, I did research this thoroughly myself, so here's a list: `ustvgo.net`, `yoursports.stream`, `giniko.com`, `sdw-net.me` `kmediatv.com` (although this just looks like a clone of `123tvgo.com`)

I keep feeling like there may be phone apps for this that i don't know about that might be easier then trying to hit sites directly.

**Comment by rleroi** [comment]
It seems that acestream links are actually just magnet links in disguise, if im correct, they just convert to: magnet:?xt=urn:btih:ACESTREAMID
I will try this out later and let you know the results.

Thanks for doing some research on the streaming sites, I'll try them out in vlc first.
It would be nice if Stremio would allows us to set custom headers.

And yes, of course it would be nicer to have an API instead of having to parse the HTML or even eval() their JS to get the stream url. ;)

**Comment by BoredLama** [comment]
I don't know if converting the acestream links to magnet links will work.. in order for live streams to work they need to have an infinite amount of data to receive, while torrents can only have a finite predefined amount of data.

You can try converting them, but I just feel like the technology itself can't support that.

My guess is that acestream uses some auto-updating server side list of magnet links, that get pulled constantly by viewers and include small segments of video, like HLS does for example. But that's just my theory on it..

**Comment by BoredLama** [comment]
> or even eval() their JS to get the stream url

also about this, i think headless browsers (like `phantomjs` or `puppeteer`) are much better then `eval()` and easier to use, but those can be hard on a server (due to more bandwidth usage / limitations of running binaries)

**Comment by rleroi** [comment]
I'll have a look at puppeteer, I was using Selenium (non headless) to scrape arconaitv links before.
Also there was an npm package safe-eval or something that will sandbox the eval() but it didn't work on first try.
